bentoml/frameworks/detectron.py

```diff
@@ -0,0 +1,147 @@
+        DetectronModelArtifact
+        InvalidArgument: invalid argument type, model being packed must be instance of
+            torch.nn.Module
+
+    Example usage:
+
+    >>> # Train model with data
+    >>>
+    >>>
+    >>> import bentoml
+    >>> from bentoml.adapters import ImageInput
+    >>> from bentoml.frameworks.detectron import DetectronModelArtifact
+    >>> from detectron2.data import transforms as T
+    >>>
+    >>> @bentoml.env(infer_pip_packages=True)
+    >>> @bentoml.artifacts([DetectronModelArtifact('model')])
+    >>> class CocoDetectronService(bentoml.BentoService):
+    >>>
+    >>>     @bentoml.api(input=ImageInput(), batch=False)
+    >>>     def predict(self, img: np.ndarray) -> Dict:
+    >>>         _aug = T.ResizeShortestEdge(
+    >>>            [800, 800], 1333
+    >>>        )
+    >>>        boxes = None
+    >>>        scores = None
+    >>>        pred_classes = None
+    >>>        pred_masks= None
+    >>>        try:
+    >>>            original_image = img[:, :, ::-1]
+    >>>            height, width = original_image.shape[:2]
+    >>>            image = _aug.get_transform(original_image).
+    >>>                             apply_image(original_image)
+    >>>            image = torch.as_tensor(image.astype("float32").
+    >>>                             transpose(2, 0, 1))
+    >>>            inputs = {"image": image, "height": height, "width": width}
+    >>>            predictions = self.artifacts.model([inputs])[0]
+    >>>
+    >>>
+    >>> cfg = get_cfg()
+    >>> cfg.merge_from_file("input_model.yaml")
+    >>> meta_arch = META_ARCH_REGISTRY.get(cfg.MODEL.META_ARCHITECTURE)
+    >>> model = meta_arch(cfg)
+    >>> model.eval()
+    >>> device = "cuda:{}".format(0)
+    >>> model.to(device)
+    >>> checkpointer = DetectionCheckpointer(model)
+    >>> checkpointer.load("output/model.pth")
+    >>> metadata = {
+    >>>     'device' : device
+    >>> }
+    >>> bento_svc = CocoDetectronService()
+    >>> bento_svc.pack('model', model, metadata, "input_model.yaml")
+    >>> saved_path = bento_svc.save()
+    """
+
+    def __init__(self, name):
+        super(DetectronModelArtifact, self).__init__(name)
+        self._file_name = name
+        self._model = None
+        self._aug = None
+        self._input_model_yaml = None
+
+    def set_dependencies(self, env: BentoServiceEnv):
+        env.add_pip_packages(['torch', "detectron2"])
+
+    def _model_file_path(self, base_path):
+        return os.path.join(base_path, self.name)
+
+    def pack(
+        self, model, metadata=None, input_model_yaml=None
+    ):  # pylint:disable=arguments-differ
+        try:
+            import detectron2  # noqa # pylint: disable=unused-import
+        except ImportError:
+            raise MissingDependencyException(
+                "Detectron package is required to use DetectronModelArtifact"
+            )
+        self._model = model
+        self._metadata = metadata
```

Comment:
We don't need to set `_metadata` here.  The `BentoServiceArtifact` class will handle it for us